### PR TITLE
allow unicode flat and sharp as input for chord symbols

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
and also double flat and double sharp, &Delta;, &deg;, &oslash; and &Oslash;.

See #3461, which is for master, while this here is for 2.2

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Unfortunatly it fails too

**Comment by MarcSabatella** [comment]
I suspect maybe you are using the wrong unicode symbols in your testing?  If you are using the F2 palette, you need the small flat that is fourth from the top left, not the larger one at top left.  The larger one is a different glyph, I think from the SMuFL area maybe?  Perhaps the code should be adjusted to handle both.

Anyhow, all I did was copy and paste your code into my copy.  It works fine for me.  Here's how I tested:

1) use default empty score
2) select first rest
3) Ctrl+K
4) Type "D Ctrl+Shift+B 7 Ctrl+Shift+B" to enter Db7b5 using real flat signs
5) Press Esc

Result: chord is converted properly.  To test further, I double click it to edit again, and it shows the flat signs have been converted to b's.

I also tested by inserted the flat signs from the F2 palette (have to insert into plain text first, then copy and paste into a chord symbol).  As mentioned, it works with the Unicode flat, not with the SMuFL flat.

**Comment by Jojo-Schmitz** [comment]
Hmm, I was testing with a copy/paste of the triangle above. Seems to have been the mistake, EBKAC :-(

**Comment by MarcSabatella** [comment]
And such is part of the problem with this - we really can't control what glyph someone might erroneously try to use. There are lots of triangles and circles out there, and at least two but actually more different flats and other accidentals.

**Comment by Jojo-Schmitz** [comment]
That 4.Type "D Ctrl+Shift+B 7 Ctrl+Shift+B" works even without my PR and converts the flat to b

**Comment by Jojo-Schmitz** [comment]
Except for `void Harmony::endEdit()` vs. ` void Harmony::endEdit(EditData& ed)` and `Text::endEdit();` vs. `TextBase::endEdit(ed);` there is no difference in the patch to #3461 (so cherry-picking should be relatively easy either way), but in master the chord symbol rendering currently doesn't work at all, so only here we can test this properly.

File: libmscore/harmony.cpp [hunk 1]
```diff
@@ -732,6 +732,14 @@ bool Harmony::edit(MuseScoreView* view, Grip grip, int key, Qt::KeyboardModifier
 
 void Harmony::endEdit()
       {
+      _userName.replace("\u1d12b", "bb"); // double-flat
+      _userName.replace("\u226d",  "b");  // flat
```
Comment (Jojo-Schmitz) [inline libmscore/harmony.cpp]: stupid mistake, it should be `\u266d`

**Comment by Jojo-Schmitz** [comment]
Taking the small ♭ (copies as `<sym>accidentalFlat</sym>`) from the F2 palette works (i.e. gets replaced with a regular `b`) even without the change (just like Ctrl+Shift+B just works), same for the small ♯ (copies as `<sym>accidentalSharp</sym>`), changes to a regular `#`. The small double flat and double sharp don't work at all though, they just vanish from the input.
The large ♭ and ♯ from the F2 palette don't work, neither with, nor without the change of this PR, they just remain the flat and sharp symbol.
Stepping thru the code seems to indicate that `_username` is just empty when entering `Harmony::endEdit`.

**Comment by MarcSabatella** [comment]
Good point.  Probably makes more sense to use raw "_text".  There could be more to it though because the text actually gets some layout-related processing.  And as I keep mentioning, it's likely to be more complex for chord symbols than ordinary text due to transposition.  Might be better to just allow these symbols as is and recognize them in parsing, and maybe add appropriate rendering for them in the chord description files.

All in all, still seems very low priority to me given there are already much easier ways, supported, commonly used in all other software, and well-documented for the user to get the desired result.

**Comment by Jojo-Schmitz** [comment]
Well, `_text` is private (to `Text` in 2.x, `TextBase` in master). so I guess it should be `Text::xmlText()` (just `xmlText()` doesn't work). But that doesn't work either.

**Comment by Jojo-Schmitz** [comment]
_allow these symbols as is and recognize them in parsing_ would be a no-go for 2.x, as that'd be incompatible with version before that change

**Comment by Jojo-Schmitz** [comment]
Ah! Now I see something, in QtCreator's Application output window:

    Warning: Data set on unsupported clipboard mode. QMimeData object will be deleted.

Happens when I select (to copy) one of the big F2 symbols (♭ or ♯). Not sure what that means though.

**Comment by MarcSabatella** [comment]
Indeed, probably not literally _text.  As mentioned, layout is complex so simply replacing characters can be tricky.

As for compatibility, good point, although I'm personally not overly concerned about that in this case either.  As far as I can tell 2.1 would still open a score created in 2.2 using these characters just fine, it just would produce some of the same glitches in rendering, transposition, etc that you already get.

**Comment by Jojo-Schmitz** [comment]
But where the heck it the raw string?

**Comment by Jojo-Schmitz** [comment]
Will resubmit for 2.3

**Pull request closed without merging**